docs/pages/management/infrastructure-as-code/terraform-starter/enroll-resources.mdx

```diff
@@ -0,0 +1,749 @@
+   `terraform-starter` module for your cloud provider:
+
+   <Tabs>
+   <TabItem label="AWS">
+   
+   ```hcl
+    module "agent_installation" {
+     source                = "./teleport"
+     agent_count           = 2
+     agent_roles           = ["Node"]
+     labels                = {}
+     proxy_service_address = "teleport.example.com:443"
+     teleport_edition      = "cloud"
+     teleport_version      = "(=teleport.version=)"
+   }
+   
+    module "agent_deployment" {
+     region           = ""
+     source           = "./cloud"
+     subnet_id        = ""
+     userdata_scripts = module.agent_installation.userdata_scripts
+   }
+   ```
+   
+   </TabItem>
+   <TabItem label="Google Cloud">
+   
+   ```hcl
+    module "agent_installation" {
+     source                = "./teleport"
+     agent_count           = 2
+     agent_roles           = ["Node"]
+     labels                = {}
+     proxy_service_address = "teleport.example.com:443"
+     teleport_edition      = "cloud"
+     teleport_version      = "(=teleport.version=)"
+   }
+   
+    module "agent_deployment" {
+     gcp_zone         = "us-east1-b"
+     google_project   = ""
+     source           = "./cloud"
+     subnet_id        = ""
+     userdata_scripts = module.agent_installation.userdata_scripts
+   }
+   ```
+   
+   </TabItem>
+   <TabItem label="Azure">
+   
+   ```hcl
+    module "agent_installation" {
+     source                = "./teleport"
+     agent_count           = 2
+     agent_roles           = ["Node"]
+     labels                = {}
+     proxy_service_address = "teleport.example.com:443"
+     teleport_edition      = "cloud"
+     teleport_version      = "(=teleport.version=)"
+   }
+   
+    module "agent_deployment" {
+     azure_resource_group = ""
+     public_key_path      = ""
+     region               = "East US"
+     source               = "./cloud"
+     subnet_id            = ""
+     userdata_scripts     = module.agent_installation.userdata_scripts
+   }
+   ```
+   
+   </TabItem>
+   </Tabs>
+
+Edit the ` module "agent_installation"` block in `main.tf` as follows:
+
+1. Assign `agent_count` to `2` for high availability. As you scale your Teleport
+   usage, you can increase this count to ease the load on each Agent. 
+
+1. Modify the elements of the `agent_roles` list.
+
+   The Teleport Auth Service associates a join token with one or more roles,
+   identifying the Teleport services that are allowed to use the token. Modify
+   `agent_roles` to include the token roles that correspond to the Teleport
+   services you plan to run on your Agent nodes:
+
+   | Role value  | Service it enables           |
+   |-------------|------------------------------|
+   | `App`       | Teleport Application Service |
+   | `Discovery` | Teleport Discovery Service   |
+   | `Db`        | Teleport Database Service    |
+   | `Kube`      | Teleport Kubernetes Service  |
```

Comment:
Setting up the kube service is more complex than passing the role and starting an agent.

You build an agent that dynamically servers multiple clusters like for apps and DBs but it's way harder and not supported by the Terraform provider currently.

The easiest way to start with app access is to deploy the `teleport-kube-agent` Helm chart.

I think we should not mention roles that this guide doesn't cover (or add the missing `WindowsDesktop`, `Okta`, and `MDM` roles, but I'm not sure how actionable this is)